python/amici/ode_export.py

```diff
@@ -1590,7 +1588,8 @@ def _derivative(self, eq: str, var: str, name: str = None) -> None:
         sym_var = self.sym(var, needs_stripped_symbols)
 
         if min(eq.shape) and min(sym_var.shape) \
-                and eq.is_zero is not True and sym_var.is_zero is not True:
+                and eq.is_zero is not True and sym_var.is_zero is not True \
+                and not sym_var.free_symbols.isdisjoint(eq.free_symbols):
```

Comment:
This (together with line 938) would enable correct Jacobian also for rate rules.
Like it!

Reply:
Yes, and it's done in 10s for the large model. Overall brings down computation time way more than I expected :D.

Reply:
Yes, that's mostly due to not computing dxdotdp_explicit any more... That saved about 10 minutes...

Reply:
Maybe good to put this whole logic into a separate function to use that everywhere in ode_export.py?